agent/src/main/java/com/teamscale/jacoco/agent/options/AgentOptions.java

```diff
@@ -340,13 +358,25 @@ public IUploader createUploader(Instrumentation instrumentation) throws Uploader
 	}
 
 	private IUploader createDelayedTeamscaleUploader(Instrumentation instrumentation) {
-		DelayedUploader<String> uploader = new DelayedUploader<>(
-				revision -> {
-					teamscaleServer.revision = revision;
+		DelayedUploader<ProjectRevision> uploader = new DelayedUploader<>(
+				projectRevision -> {
+					if (!StringUtils.isEmpty(projectRevision.getProject()) && !teamscaleServer.project.equals(projectRevision.getProject())) {
+						logger.warn("Teamscale project specified in the agent configuration is not the same as the Teamscale project specified in git.properties file(s). Proceeding to upload to the" +
+								" Teamscale project specified in the agent configuration.");
```

Comment:
I would add the name of the concrete project to which we upload and the project we ignored as that might help to easier spot the misconfiguration